test/ehttpc_tests.erl

```diff
@@ -487,6 +487,88 @@ cool_down_after_5_reqs_test() ->
         end
     ).
 
+zombie_detect_inflight_not_full_test() ->
+    Port = ?PORT,
+    ServerOpts = #{
+        port => Port,
+        name => ?FUNCTION_NAME,
+        %% no response during this test
+        delay => 30_000,
+        oneoff => false
+    },
+    PoolOpts0 = pool_opts("127.0.0.1", Port, _Pipelining = 5, _PrioritiseLatest = false),
+    PoolOpts = [{max_inactive, 1_000} | PoolOpts0],
+    Reqs = [["test-put-", integer_to_list(I)] || I <- lists:seq(1, 4)],
+    ?WITH(
+        ServerOpts,
+        PoolOpts,
+        begin
+            lists:foreach(
+                fun(Req) ->
+                    spawn_link(fun() -> ehttpc:request(?POOL, put, {<<"/">>, [], Req}, 100) end)
+                end,
+                Reqs
+            ),
+            Pid = ehttpc_pool:pick_worker(?POOL),
+            {ok, _} = ?block_until(#{?snk_kind := reconnect}, 2500, infinity),
+            %% let the EXIT signal get to ehttpc process
+            timer:sleep(100),
+            #{requests := Requests} = ehttpc:get_state(Pid, normal),
```

Comment:
Nit: maybe we could block on `handle_client_down` so that there's no need to sleep?

```suggestion
            {ok, _} = ?block_until(#{?snk_kind := reconnect}, 2500, infinity),
            {ok, _} = ?block_until(#{?snk_kind := handle_client_down}),
            #{requests := Requests} = ehttpc:get_state(Pid, normal),
```